app/controller.go

```diff
@@ -214,27 +200,34 @@ func listSnapshot(c *controller.Controller) ([]string, error) {
 func autoDeleteSnapshot(c *controller.Controller) error {
 	c.Lock()
 	replicas := c.ListReplicas()
-	c.Unlock()
 	if err := checkPrerequisites(c, replicas); err != nil {
+		c.Unlock()
 		return err
 	}
+	c.IsSnapDeletionInProgress = true
+	c.Unlock()
+	defer func() {
+		c.Lock()
+		c.IsSnapDeletionInProgress = false
+		c.Unlock()
+	}()
+	reps := []rest.Replica{}
+	for _, rep := range replicas {
+		r := rest.Replica{
+			Address: rep.Address,
+			Mode:    string(rep.Mode),
+		}
+		reps = append(reps, r)
+	}
```

Comment:
Unlock needs to be done over here instead of above since replicas slice can be updated by some other goroutine

Reply:
not required since copying the slice in new approach